### ISSUE TITLE
Provider crash when applying Velero Chart

### ISSUE THREAD

**Issue by djsunn** [hashicorp/terraform-provider-helm#579]
### Terraform Version and Provider Version
0.13.2

### Affected Resource(s)
- helm_release

### Terraform Configuration Files
```hcl
resource "helm_release" "velero" {
  depends_on = [azurerm_storage_account.velero]

  repository = "https://vmware-tanzu.github.io/helm-charts"
  chart      = "velero"
  name       = "velero"

  namespace = kubernetes_namespace.velero_namespace.metadata[0].name

  timeout = var.helm_timeout

  values = [
    file("./config/backup/velero.yaml"),
    file("./config/backup/schedule.yaml")
  ]

  set {
    name  = "initContainers.name[0].image"
    value = "velero/velero-plugin-for-microsoft-azure:${var.velero_azure_plugin_version}"
  }

  set {
    name  = "configuration.backupStorageLocation.config.resourceGroup"
    value = azurerm_resource_group.backup.name
  }

  set {
    name  = "configuration.backupStorageLocation.config.subscriptionId"
    value = var.subscription_id
  }

  set {
    name  = "configuration.backupStorageLocation.config.storageAccount"
    value = azurerm_storage_account.velero.name
  }

  set {
    name  = "configuration.volumeSnapshotLocation.config.resourceGroup"
    value = azurerm_resource_group.backup.name
  }

  set {
    name  = "configuration.volumeSnapshotLocation.config.subscriptionId"
    value = var.subscription_id
  }

  set_sensitive {
    name  = "credentials.secretContents.AZURE_SUBSCRIPTION_ID"
    value = var.subscription_id
  }

  set_sensitive {
    name  = "credentials.secretContents.AZURE_TENANT_ID"
    value = var.tenant_id
  }

  set_sensitive {
    name  = "credentials.secretContents.AZURE_CLIENT_ID"
    value = var.client_id
  }

  set_sensitive {
    name  = "credentials.secretContents.AZURE_CLIENT_SECRET"
    value = var.client_secret
  }

  set_sensitive {
    name  = "credentials.secretContents.AZURE_RESOURCE_GROUP"
    value = azurerm_resource_group.backup.name
  }

  set {
    name  = "schedules.rbdi.namespaces[0]"
    value = data.kubernetes_namespace.datahub_system_namespace.metadata[0].name
  }

  set {
    name  = "schedules.rbdi.namespaces[1]"
    value = kubernetes_namespace.ingress_controller_namespace.metadata[0].name
  }

  set {
    name  = "schedules.rbdi.namespaces[2]"
    value = kubernetes_namespace.monitoring_namespace.metadata[0].name
  }

  set {
    name  = "schedules.rbdi.namespaces[3]"
    value = data.kubernetes_namespace.slcbridge_namespace.metadata[0].name
  }

  set {
    name  = "schedules.rbdi.namespaces[4]"
    value = kubernetes_namespace.datahub_namespace.metadata[0].name
  }
}
```

### Debug Output
https://gist.github.com/djsunn/6ee9031223aa973ef1bee7542d2bb081
(Updated, previous output was incomplete)

NOTE: In addition to Terraform debugging, please set HELM_DEBUG=1 to enable debugging info from helm.-->

### Panic Output
panic: interface conversion: interface {} is []interface {}, not map[string]interface {}

### Expected Behavior
Deploy heml_release resource or provide meaningful error message

### Actual Behavior
Provider crash

### Steps to Reproduce
1. `terraform apply`

### Important Factoids
-

### References
-

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by djsunn** [comment]
The previous crash seems to be caused by this being set wrongly:

```hcl
  set {
    name  = "initContainers.name[0].image"
    value = "velero/velero-plugin-for-microsoft-azure:${var.velero_azure_plugin_version}"
  }
```

Which should be this:
```hcl
  set {
    name  = "initContainers[0].name.image"
    value = "velero/velero-plugin-for-microsoft-azure:${var.velero_azure_plugin_version}"
  }
```

Now i get this crash:
https://gist.github.com/djsunn/82763cf0f031d58baf332372eff08139

**Comment by jrhouston** [comment]
Thanks for opening this @djsunn – can you update to the latest version of this provider and try this again? This crash, at least, seems to have been fixed in the latest helm, which we added in v1.3.1: https://github.com/helm/helm/blob/master/pkg/strvals/parser.go#L156

**Comment by jrhouston** [comment]
I'm going to close this as there was no follow up.
